src/mcp/tool.ts

```diff
@@ -0,0 +1,31 @@
+import { CallToolResult } from "@modelcontextprotocol/sdk/types";
+import { z, ZodTypeAny } from "zod";
+import { zodToJsonSchema } from "zod-to-json-schema";
+
+export interface ServerTool<InputSchema extends ZodTypeAny = ZodTypeAny> {
+  mcp: {
+    name: string;
+    description?: string;
+    inputSchema: any;
```

Comment:
Could probably make this more strongly typed. It looks like this is expected to be JSON schema, so  we could either roll our own limited version of that, or use https://github.com/StefanTerdell/zod-to-json-schema/blob/master/src/parseTypes.ts#L58 (i think it is exported?)

Reply:
Yeah, ideally this would have a slightly better type but the MCP SDK has its own idea of what the type is that isn't exported well and fundamentally this is an internal implementation detail so I'm going to leave it as-is for now.